src/elements/content-sidebar/__tests__/MetadataInstanceEditor.test.tsx

```diff
@@ -63,4 +87,39 @@ describe('MetadataInstanceEditor', () => {
         const unsavedChangesModal = screen.getByText('Unsaved Changes');
         expect(unsavedChangesModal).toBeInTheDocument();
     });
+
+    test('Should call onCancel when canceling editing', () => {
+        const props: MetadataInstanceEditorProps = { ...defaultProps, template: mockCustomMetadataTemplate };
+        const { getByRole } = render(<MetadataInstanceEditor {...props} />);
+
+        act(() => {
+            fireEvent.click(getByRole('button', { name: 'Cancel' }));
+        });
+
+        expect(mockOnCancel).toHaveBeenCalled();
+    });
+
+    test('Should call onUnsavedChangesModalCancel instead onCancel when canceling through UnsavedChangesModal', () => {
+        const props: MetadataInstanceEditorProps = { ...defaultProps, template: mockCustomMetadataTemplateWithField };
+        const { getByRole, rerender } = render(<MetadataInstanceEditor {...props} />);
+        const input = getByRole('textbox');
```

Comment:
```suggestion
        const input = await findByRole('textbox');
```
maybe in order to get rid of the `act`